apply-ParagraphStandards.py

```diff
@@ -0,0 +1,382 @@
+        # print help information and exit:
+        print(err)  # will print something like "option -a not recognized"
+        usage()
+        sys.exit(1)
+
+    with open("./NotebookSanityExceptions.yaml", 'r') as stream:
+        sanity_exceptions = yaml.load(stream)
+
+    track_vars = {}
+    track_vars['modify'] = False
+    track_vars['tests'] = 'All'
+    track_vars['has_changes'] = False
+    track_vars['para_count'] = 1
+    track_vars['notebook_id'] = ""
+    track_vars['status'] = 0
+
+    for options, option_args in parsed_options:
+        if options in ("-h", "--help"):
+            usage()
+            sys.exit(0)
+        elif options in ("-t", "--test"):
+            if option_args.lower() == 'all':
+                track_vars['tests'] = 'All'
+            elif option_args.lower() == 'mdtimecheck':
+                track_vars['tests'] = 'MDTC'
+            elif option_args.lower() == 'codeblockfix':
+                track_vars['tests'] = 'CBF'
+        elif options in ("-m", "--modify"):
+            track_vars['modify'] = True
+        else:
+            assert False, "unhandled option: " + options + \
+                ", option arguments (" + option_args + ")"
+        if len(parsed_args) > 0:
+            print("stand alone arguments have been ignored: ")
+            for ignored_arg in parsed_args:
+                print("\t" + ignored_arg)
+
+    notefiles = glob.glob('./**/note.json')
+
+    for note_file in notefiles:
+        track_vars['has_changes'] = False
+        with open(note_file, 'r+') as file_handler:
+            track_vars['notebook_id'] = note_file.split('/')[1]
+            data = json.load(file_handler)
+            # We will start at normal counting, so when you load the notebook,
+            # you can simply count to match with the reported paragraph number.
+            # some notebook paragraphs have a 'title' set, at some point when
+            # that is more populated we can switch to referring to the
+            # paragraphs by title instead throughout.  For now paragraph
+            # counting will have to be sufficient.
+            track_vars['para_count'] = 1
+            for notebook_paragraph in data['paragraphs']:
+                test_paragraph(notebook_paragraph,
+                               track_vars, sanity_exceptions)
+                track_vars['para_count'] = track_vars['para_count'] + 1
+            if track_vars['modify'] and track_vars['has_changes']:
+                print("Saving " + track_vars['notebook_id'])
+                try:
+                    file_handler.seek(0)
+                    json.dump(data, file_handler, indent=2, sort_keys=False)
+                    file_handler.truncate()
+                except:
+                    print("An error occurred while writing to " + note_file)
+                    track_vars['status'] = 1
+    return track_vars['status']
+
+
+# Call the main Routine
+exitcode = 0
+exitcode = main(sys.argv[1:])
+sys.exit(exitcode)
```

Comment:
You can combine the lines above to simply `sys.exit(main(sys.argv[1:]))`. If you don't want to do that, at least remove the unnecessary assignment `exitcode = 0`.